script/set_channel.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
```

Comment:
If you don't hoist the logic up to the top per my previous suggestion, instead of creating a new bash file you could also instead add a [repeated node](https://yaml.org/spec/1.2/spec.html#id2760395) template to .cirrus.yml so it's in the same file.  Same comment though about needing to add `CHANNEL: "master"` in the few places missing it.
```yaml
set_channel_template: &SET_CHANNEL_TEMPLATE
  script:
    - flutter channel $CHANNEL
    - flutter upgrade
...
  matrix:
    ### Platform-agnostic tasks ###
    - name: plugin_tools_tests
      << : *SET_CHANNEL_TEMPLATE
...
    - name: build_all_plugins_apk
      env:
        matrix:
          CHANNEL: "master"
          CHANNEL: "stable"
      << : *SET_CHANNEL_TEMPLATE
```
etc.

Reply:
I ended up doing both, for even less duplication :)